plugins/modules/sns_topic_info.py

```diff
@@ -0,0 +1,151 @@
+  returned: suceess
+  type: list
+  contains:
+    sns_arn:
+        description: The ARN of the topic.
+        type: str
+        returned: always
+        sample: "arn:aws:sns:us-east-2:111111111111:my_topic_name"
+    sns_topic:
+        description: Dict of sns topic details
+        type: complex
+        returned: always
+        contains:
+            delivery_policy:
+                description: Delivery policy for the SNS topic
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: {
+                    "http":{"defaultHealthyRetryPolicy":{"minDelayTarget":20,"maxDelayTarget":20,"numRetries":3,"numMaxDelayRetries":0,
+                    "numNoDelayRetries":0,"numMinDelayRetries":0,"backoffFunction":"linear"},"disableSubscriptionOverrides":false}
+                }
+            display_name:
+                description: Display name for SNS topic
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: My topic name
+            owner:
+                description: AWS account that owns the topic
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: '111111111111'
+            policy:
+                description: Policy for the SNS topic
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: {
+                    "Version":"2012-10-17","Id":"SomePolicyId","Statement":[{"Sid":"ANewSid","Effect":"Allow","Principal":{"AWS":"arn:aws:iam::111111111111:root"},
+                    "Action":"sns:Subscribe","Resource":"arn:aws:sns:us-east-2:111111111111:ansible-test-dummy-topic","Condition":{"StringEquals":{"sns:Protocol":"email"}}}]
+                }
+            subscriptions:
+                description: List of subscribers to the topic in this AWS account
+                returned: always
+                type: list
+                sample: []
+            subscriptions_confirmed:
+                description: Count of confirmed subscriptions
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: '0'
+            subscriptions_deleted:
+                description: Count of deleted subscriptions
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: '0'
+            subscriptions_pending:
+                description: Count of pending subscriptions
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: '0'
+            topic_arn:
+                description: ARN of the SNS topic (equivalent to sns_arn)
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: arn:aws:sns:us-east-2:111111111111:ansible-test-dummy-topic
```

Comment:
Info module also returns `subscriptions_added`, `subscriptions_existing`, `subscriptions_new`, `subscriptions_purge`, and `topic_type`. Do we want to include these as well?

Reply:
Added.